data-models/api/v1alpha1/healthstatus_types.go

```diff
@@ -0,0 +1,135 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:validation:Enum=NotStarted;InProgress;Failed;Succeeded;AlreadyDrained;UnQuarantined;Quarantined;AlreadyQuarantined;Cancelled
+type Status string
+
+const (
+	StatusNotStarted Status = "NotStarted"
+	StatusInProgress Status = "InProgress"
+	StatusFailed     Status = "Failed"
+	StatusSucceeded  Status = "Succeeded"
+	AlreadyDrained   Status = "AlreadyDrained"
+
+	UnQuarantined      Status = "UnQuarantined"
+	Quarantined        Status = "Quarantined"
+	AlreadyQuarantined Status = "AlreadyQuarantined"
+	Cancelled          Status = "Cancelled"
+)
+
+// HealthEventSpec defines the desired state of HealthStatus
+type HealthEventSpec struct {
```

Comment:
We don't want to manage two copies of the same struct just to make it a kubernetes type.

Can we point directly to the objects that already exist? So it would look something like this

```
// +kubebuilder:object:root=true
// +kubebuilder:subresource:status
type HealthEvent struct {
  Spec HealthEvent // this is the existing health event object
  Status HealthEventStatus // this is the existing status object
}
```